### DISCUSSION TITLE
How to get values from frontend script

### DISCUSSION THREAD

**Discussion by SergeyPriem** [writer/writer-framework#466] [General]
I want to get the screen size from the Javascript function and use it in Python functions.

The function  **state.call_frontend_function** returns None.

How to get values from the frontend module?

**Comment by ramedina86** [comment]
Hi Sergey, the architecture, due to security concerns, only allows communication from client to server via events.

You can generate an event and catch it on the other side.

I tested this and it works as expected. Then the event `myinfo` can be caught from the _Root_ component by setting the appropriate handler.

```js
export function sendCustomEvent() {
	alert("Sending");
	const ev = new CustomEvent("myinfo", {
		detail: {
			payload: window.screen.height,
			callback: () => {
				alert("Custom event sent successfully.");
			},
		},
	});
	globalThis.core.forwardEvent(ev, [{componentId: "root", instanceNumber: 0}], true);
}
```

What's the use case by the way? We may look into implementing a more straightforward way of doing this.

**Reply by SergeyPriem** [reply]
Hello Ramiro,

Thank you for your response. The goal is to obtain the screen size and adjust the user interface accordingly for different devices such as tablets, phones, and desktops. This may include altering the menu/sidebar or positioning the footer at the bottom (or adjust the size of modal window). Additionally, we may explore other ideas that leverage this capability in the future.

**Reply by SergeyPriem** [reply]
Hi, Ramiro!

I don't understand which handler to use to catch the event from JS. If you have time to explain more precisely, it would be great. 
For now, I found a simple solution for UI adjustment (through CSS: @media (min-width: 1200px) {...}). And now I understand your surprise when I ask about such functionality.

I'm still learning))

**Comment by pandadezzz** [comment]
@ramedina86 
i also would like this feature. i am trying to get geolocation from user. 
something like this:

``` js
export function getLocationAndShowCoordinates() {
  if (navigator.geolocation) {
    navigator.geolocation.getCurrentPosition(function(position) {
      console.log("Latitude: " + position.coords.latitude);
      console.log("Longitude: " + position.coords.longitude);

      document.getElementById('coordinates').innerHTML = 
        "Latitude: " + position.coords.latitude + "<br>" +
        "Longitude: " + position.coords.longitude;
    });
  } else {
    x.innerHTML = "Geolocation is not supported by this browser.";
  }
}
```


I now don't know how and where to catch the event. when you say root component is that src/ui/src/components/core/root
/CoreRoot.vue ? novice here. 

Thanks!

**Reply by FabienArcellier** [reply]
Here is the process

![Peek 2024-09-19 07-08](https://github.com/user-attachments/assets/1031f9eb-1c27-468c-bf59-1e6fbd9a1068)

**Reply by pandadezzz** [reply]
thanks so much! it's a lot simpler than i thought haha. really opens up to a lot more flexibility
